app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -20,10 +20,10 @@ module TwoFactorAuthenticatable
   private
 
   def apply_secure_headers_override
-    return unless after_sign_in_path_for.start_with?(openid_connect_authorize_path)
+    return unless stored_url_for_user&.start_with?(openid_connect_authorize_path)
```

Comment:
to check my understanding here, the reason we are using `stored_url_for_user` and accessing directly is because `after_sign_in_path_for` would delete the value?